conformance/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -154,4 +139,33 @@ object DynamicInterpreter {
     }
   }
 
+  private def getStepsAndExplode(conformance: ConfDataset, inputDf: Dataset[Row], experimentalMappingRule: Boolean):
```

Comment:
This method seems having several unrelated responsibilities. Consider making each statement of this function a separate method, e.g.
```
getOrderedSteps(...)
addErrorColumnIfMissing(...)
explodeAllArrays(...)
```

Then, you can call each method one by one from `interpret()` or you can leave `getStpsAndExplode()` as a grouping method. Up to you.

Reply:
I understand that this refactor has left much to the desired. I will rework it to include you remarks.